### PR TITLE
`Development`: Unify logger usage

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context
This PR unifies the logger usage accross Artemis. 

### Description
- Made all logger variables `private static final`
- Always use the name `log` (instead of `logger` or `LOGGER`)
- Removed string concatenation in logger calls

### Steps for Testing
code review.
This PR is best reviewed by looking at each commit:
- c40a6eddc95296e2f540a8ad1717712975cbf1d3 changed the fields to `static final` using an IntelliJ inspection
- f833fc28f506b7465b3e2efa8b180408b3839f5c changed the names to `log`
- 30a66f295db530a9546b175c00720493c2dc4ff3 removed string concatenations in logger calls.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/domain/notification/Notification.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
 public abstract class Notification extends DomainObject {
 
     @Transient
```
Comment (muradium) [inline src/main/java/de/tum/in/www1/artemis/domain/notification/Notification.java:39]: Do we still need Logger fields to be Transient, as they're static final now?

**Review by muradium — COMMENTED** [review]
LGTM. Left a small comment about transient fields.

**Review by aplr — DISMISSED** [review]
Code LGTM

**Review by egekurt123 — DISMISSED** [review]
Code changes LGTM

**Review by coolchock — DISMISSED** [review]
code lgtm

**Review by Hialus — APPROVED** [review]
Changes look good to me.
Also thanks for adding an architecture test to check for this automatically 🚀

**Review by krusche — APPROVED** [review]

**Pull request merged**